File: Java/Greeting.java
import java.io.Console ;

public class Greeting {
    public static void main( String[] args ) {
        // Get the system console
        Console console = System.console() ;

        // Ask for the users name, and load the response into `name`
        String name = console.readLine( "Enter your name: " ) ;

        // And print our greeting
        System.out.printf( "Hello %s\n", name ) ;
    }
}

File: Cpp/greeting.cpp
#include <string>
#include <iostream>

int main()
{
    // Declare a String to hold our name
    std::string name ;

    // Print our question    
    std::cout << "Enter your name: " ;

    // Then read the response from the user into `name`
    std::getline( std::cin, name ) ;

    // And print our greeting
    std::cout << "Hello " << name << std::endl ;

    // Exit cleanly
    return 0 ;
}
